src/openms/include/OpenMS/DATASTRUCTURES/FASTAContainer.h

```diff
@@ -360,141 +372,153 @@ class DecoyHelper
 
       if (freq_prefix >= 0.8 && freq_prefix_in_proteins >= 0.4)
       {
-        if (prefix_suffix_counts.first != all_prefix_occur)
+        if (prefix_suffix_counts.first != decoy_stats.all_prefix_occur)
         {
           OPENMS_LOG_WARN << "More than one decoy prefix observed!" << std::endl;
           OPENMS_LOG_WARN << "Using most frequent decoy prefix (" << (int)(freq_prefix * 100) << "%)" << std::endl;
         }
 
-        return { true, decoy_case_sensitive[case_insensitive_decoy_string], true};
+        return { true, decoy_stats.decoy_case_sensitive[case_insensitive_decoy_string], true};
       }
     }
 
     // Decoy suffix occurred at least 80% of all suffixes + observed in at least 40% of all proteins -> set it as suffix decoy
-    for (const auto& pair : decoy_count)
+    for (const auto& pair : decoy_stats.decoy_count)
     {
       const std::string& case_insensitive_decoy_string = pair.first;
       const std::pair<Size, Size>& prefix_suffix_counts = pair.second;
-      double freq_suffix = static_cast<double>(prefix_suffix_counts.second) / static_cast<double>(all_suffix_occur);
-      double freq_suffix_in_proteins = static_cast<double>(prefix_suffix_counts.second) / static_cast<double>(all_proteins_count);
+      double freq_suffix = static_cast<double>(prefix_suffix_counts.second) / static_cast<double>(decoy_stats.all_suffix_occur);
+      double freq_suffix_in_proteins = static_cast<double>(prefix_suffix_counts.second) / static_cast<double>(decoy_stats.all_proteins_count);
 
       if (freq_suffix >= 0.8 && freq_suffix_in_proteins >= 0.4)
       {
-        if (prefix_suffix_counts.second != all_suffix_occur)
+        if (prefix_suffix_counts.second != decoy_stats.all_suffix_occur)
         {
           OPENMS_LOG_WARN << "More than one decoy suffix observed!" << std::endl;
           OPENMS_LOG_WARN << "Using most frequent decoy suffix (" << (int)(freq_suffix * 100) << "%)" << std::endl;
         }
 
-        return { true, decoy_case_sensitive[case_insensitive_decoy_string], false};
+        return { true, decoy_stats.decoy_case_sensitive[case_insensitive_decoy_string], false};
       }
     }
 
     OPENMS_LOG_ERROR << "Unable to determine decoy string and its position. Please provide a decoy string and its position as parameters." << std::endl;
     return {false, "?", true};
   }
-  
+
+  /**
+  @brief Function to count the occurrences of decoy strings in a given set of protein names
+
+  For tested decoy strings see DecoyHelper::affixes.
+  Returns all data needed for interpretation (see DecoyHelper::DecoyStatistics).
+  */
+  template<typename T>
+  static DecoyStatistics countDecoys(FASTAContainer<T>& proteins)
+  {
+    // common decoy strings in FASTA files
+    // note: decoy prefixes/suffices must be provided in lower case
+
+    DecoyStatistics ds;
+    ds.all_prefix_occur = 0;
+    ds.all_suffix_occur = 0;
+    ds.all_proteins_count = 0;
```

Comment:
improvement:
initialize the integer members in the struct with ` = 0` directly.
Then this manual initialization here becomes unnecessary and (more importantly) this would be need for EVERY instanciation of a `DecoyStatistics`. Thus, if forgotten, it would result in weird counts.